File: frontend/src/app/core/models/response/account/account-response.ts
import {CurrencyType} from '../../enum/currency-type';
import {AccountType} from '../../enum/account-type';

export interface AccountResponse {
  id: Number;
  accountNumber: string;
  ibanCode: string;
  type: AccountType;
  currency: CurrencyType;
  balance: number;
  active: boolean;
  lastModifiedDate: string;
}

File: src/main/java/com/example/dto/response/account/AccountResponse.java
package com.example.dto.response.account;

import java.math.BigDecimal;

public record AccountResponse(Long id, String accountNumber, String ibanCode, com.example.util.enums.AccountType type, com.example.util.enums.CurrencyType currency,
                              BigDecimal balance, boolean active, String lastModifiedDate) {
}